GA blog first draft (#2297)
OpenLiberty/blogs: posts/2022-06-07-microprofile-graphql-2.adoc

---
layout: post

> title: "MicroProfile GraphQL 2, significant bug fixes, and security vulnerability (CVE) fixes in Open Liberty 22.0.0.6"

Reviewer: ```suggestion
title: "MicroProfile GraphQL 2.0, security vulnerability (CVE) and bug fixes in Open Liberty 22.0.0.6"
```

# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/ryan-storey
author_github: https://github.com/ryan-storey

> seo-title: MicroProfile GraphQL 2, significant bug fixes, and security vulnerability (CVE) fixes in Open Liberty 22.0.0.6 - OpenLiberty.io

Reviewer: ```suggestion
seo-title: MicroProfile GraphQL 2.0, security vulnerability (CVE) and bug fixes in Open Liberty 22.0.0.6 - OpenLiberty.io
```

seo-description: Open Liberty 22.0.0.6 offers MicroProfile GraphQL 2.0, which incorporates Jakarta EE 9.1 dependencies, allowing the use of updated Jakarta components such as CDI 3.0, Jakarta REST 3.0, JSON-B 2.0 and more. This release also contains a number of significant bug fixes as well as important security vulnerability (CVE) fixes.
blog_description: Open Liberty 22.0.0.6 offers MicroProfile GraphQL 2.0, which incorporates Jakarta EE 9.1 dependencies, allowing the use of updated Jakarta components such as CDI 3.0, Jakarta REST 3.0, JSON-B 2.0 and more. This release also contains a number of significant bug fixes as well as important security vulnerability (CVE) fixes.
open-graph-image: https://openliberty.io/img/twitter_card.jpg
---

> = MicroProfile GraphQL 2, significant bug fixes, and security vulnerability (CVE) fixes in Open Liberty 22.0.0.6

Reviewer: ```suggestion
= MicroProfile GraphQL 2.0, security vulnerability (CVE) and bug fixes in Open Liberty 22.0.0.6
```

Ryan Storey <https://github.com/ryan-storey>
:imagesdir: /
:url-prefix:
:url-about: /
//Blank line here is necessary before starting the body of the post.

Open Liberty 22.0.0.6 offers MicroProfile GraphQL 2.0, which incorporates Jakarta EE 9.1 dependencies, allowing the use of updated Jakarta components such as CDI 3.0, Jakarta REST 3.0, JSON-B 2.0 and more. This release also contains a number of significant bug fixes as well as important security vulnerability (CVE) fixes.

In link:{url-about}[Open Liberty] 22.0.0.6:

* <<graphql, MicroProfile GraphQL 2.0>>

> View the list of fixed bugs in link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A22.0.0.6_NO_PERIODS+label%3A%22release+bug%22[22.0.0.6].

Reviewer: ```suggestion
View the list of fixed bugs in link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A22006+label%3A%22release+bug%22[22.0.0.6].
```

[#run]
== Run your apps using 22.0.0.6

If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:

[source,xml]
----
<dependency>
    <groupId>io.openliberty</groupId>
    <artifactId>openliberty-runtime</artifactId>
    <version>22.0.0.6</version>
    <type>zip</type>
</dependency>
----

Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:

[source,gradle]
----
dependencies {
    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[22.0.0.6,)'
}
----

Or if you're using Docker:

[source]
----
FROM open-liberty
----

Or take a look at our link:{url-prefix}/downloads/[Downloads page].

[link=https://stackoverflow.com/tags/open-liberty]
image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]

[#graphql]
=== MicroProfile GraphQL 2.0

> MicroProfile GraphQL 2.0 incorporates Jakarta EE 9.1 dependencies.  This allows developers to continue using the same functionality provided by MicroProfile GraphQL 1.0 but with the updated Jakarta components such as CDI 3.0, Jakarta REST 3.0, JSON-B 2.0, etc. Functionally, version 2.0 behaves the same as 1.0 because it uses the same version of the underlying implementation link:https://github.com/smallrye/smallrye-graphql[SmallRye GraphQL].  If you'd like to learn more about MicroProfile GraphQL, see our previous link:{url-prefix}/blog/2020/06/10/microprofile-graphql-open-liberty.html[blog post about MP GraphQL 1.0] and our link:{url-prefix}/blog/2020/08/28/graphql-apis-open-liberty-20009.html[blog post on how to use the GraphiQL UI and third-party Client APIs].

Reviewer: ```suggestion
MicroProfile GraphQL 2.0 incorporates Jakarta EE 9.1 dependencies.  This allows developers to continue using the same functionality provided by MP GraphQL 1.0 but with the updated Jakarta features such as link:{url-prefix}/docs/latest/reference/feature/cdi-3.0.html[CDI 3.0], link:{url-prefix}/docs/latest/reference/feature/restfulWS-3.0.html[restfulWS 3.0], link:{url-prefix}/docs/latest/reference/feature/jsonb-2.0.html[JSON-B 2.0], etc. Functionally, version 2.0 behaves the same as 1.0 because it uses the same version of the underlying implementation (link:https://github.com/smallrye/smallrye-graphql[SmallRye GraphQL]).  If you'd like to learn more about MicroProfile GraphQL, see our previous link:{url-prefix}/blog/2020/06/10/microprofile-graphql-open-liberty.html[blog post about MicroProfile GraphQL 1.0] and our link:{url-prefix}/blog/2020/08/28/graphql-apis-open-liberty-20009.html[blog post on how to use the GraphiQL UI and third-party Client APIs].

To enable the MicroProfile GraphQL 2.0 feature, add it to your `server.xml`:

[source, xml]
----
<featureManager>
  <feature>mpGraphQL-2.0</feature>
</featureManager>
----
```
